packages/web-components/src/components/ic-title-bar/ic-title-bar.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Prop,
+  h,
+} from "@stencil/core";
+import { IcSelectCustomEvent } from "../../interface";
+import {
+  IcDataTableDensityOptions,
+  IcDensityUpdateEventDetail,
+} from "../ic-data-table/ic-data-table.types";
+import { isSlotUsed } from "../../utils/helpers";
+
+const DEFAULT_TITLE_BAR_HEADER = "Title Bar";
+
+/**
+ * @slot primary-action - Render an interactive element that will perform a primary action. Renders to the right of the density select.
+ * @slot custom-actions - Render additional custom actions to the left of the density select.
+ * @slot header - Render an alternative header in the header section.
+ * @slot description - Render an alternative description in the description section.
+ */
+@Component({
+  tag: "ic-title-bar",
+  styleUrl: "ic-title-bar.css",
+  shadow: true,
+})
+export class TitleBar {
+  private initialDensitySelectOption: IcDataTableDensityOptions = "default";
+  private isEmbedded: boolean = false;
+
+  @Element() el: HTMLIcTitleBarElement;
+
+  /**
+   * The description that is displayed below the `header` and `metadata`. Can be overridden with the `description` slot.
+   */
+  @Prop() description?: string;
+
+  /**
+   * The header of the title bar. Can be overridden with the `header` slot. If used with an ic-data-table it will default to the table's `caption` unless overridden.
+   */
+  @Prop() header?: string = DEFAULT_TITLE_BAR_HEADER;
```

Comment:
```suggestion
  @Prop() heading?: string = DEFAULT_TITLE_BAR_HEADING;
```
Think it would be good for some of these props to align with page header for consistency and better DX